config/opal_check_cuda.m4

```diff
@@ -59,8 +59,8 @@ AC_ARG_WITH([cuda-libdir],
                             [Search for CUDA libraries in DIR])],
             [],
             [AS_IF([test -d "$with_cuda"],
-             [with_cuda_libdir=$(dirname $(find $with_cuda -name libcuda.so) 2> /dev/null)],
-             [with_cuda_libdir=$(dirname $(find /usr/local/cuda -name libcuda.so) 2> /dev/null)])
+             [with_cuda_libdir=$(dirname $(find $with_cuda -name libcuda.so 2> /dev/null) 2> /dev/null)],
+             [with_cuda_libdir=$(dirname $(find /usr/local/cuda -name libcuda.so 2> /dev/null) 2> /dev/null)])
```

Comment:
Per https://github.com/open-mpi/ompi/pull/12382#discussion_r1509227373, let's continue the conversation here.

Ok, so redirecting stderr here to avoid the error if `/usr/local/cuda` doesn't exist.

What if a user does `./configure --with-cuda=/some/dir/that/does/not/exist ...`, and therefore it falls over to this test and it checks `/usr/local/cuda`, instead.  Down below, we might end up emitting a WARN message about `/usr/local/cuda` instead of whatever the value of `$with_cuda` or `$with_cuda_libdir` is.  That would be quite confusing.

I don't think we want to do this kind of tomfoolery.  If the user asks us to check in `/some/directory/that/does/not/exist`, we should error and tell them that it does not exist, and they should fix their `configure` command line parameters.  We shouldn't presume to know what they actually want.

Reply:
It should do what you're describing--if the user gives `./configure --with-cuda=/some/dir/that/does/not/exist`, then `find` will return nothing, and `$with_cuda_libdir` will be empty. But really the important part is 

```
      [AS_IF([test "$with_cuda" = "yes"],
             [AS_IF([test "x`ls /usr/local/cuda/include/cuda.h 2> /dev/null`" = "x"],
                     // fail if not found in /usr/local/cuda
             [AS_IF([test ! -d "$with_cuda"],          <---- fail if $with_cuda does not exist
                    [AC_MSG_RESULT([not found])
                     AC_MSG_WARN([Directory $with_cuda not found])
                     AC_MSG_ERROR([Cannot continue])],
```

So either way, it will fail and say `/some/directory/that/does/not/exist` not found.

Reply:
Ok, I see what you're saying now -- I think I was mixing up `$with_cuda` and `$with_cuda_libdir`.  Got it.